addon/bootstrap.js

```diff
@@ -59,7 +61,59 @@ const appStartupObserver = {
   }
 }
 
+const LibraryButton = {
+  ITEM_ID: "appMenu-library-screenshots",
+
+  init(webExtension) {
+    this._initialized = true;
+    let permissionPages = [...webExtension.extension.permissions].filter(p => (/^https?:\/\//i).test(p));
+    this.PAGE_TO_OPEN = permissionPages.length ? permissionPages[0].replace(/\*$/, "") : "https://screenshots.firefox.com/";
+    this.PAGE_TO_OPEN += "shots";
+    this.ICON_URL = webExtension.extension.getURL("icons/icon-16-v2.svg");
+    this.ICON_URL_2X = webExtension.extension.getURL("icons/icon-32-v2.svg");
```

Comment:
> I'm not sure how the photon library UI works (I haven't seen mocks or specs), but right now, in the toolbar button, we show a special icon to users who haven't tried Screenshots before, and we swap in the regular icon after they complete onboarding. (Note that startBackground runs first, lazy-loading the rest of the webextension, including main.js.)

https://mozilla.invisionapp.com/share/ENBBK0F9U#/screens/229252053

<img width="337" alt="Screenshot of mock" src="https://user-images.githubusercontent.com/375983/28415079-1dbbc272-6d46-11e7-8773-37218541b86c.png">

> If this UI is replacing the toolbar button UI, I think we'd want to keep that special icon behavior for new users.

It's not replacing the toolbar button. We will replace the toolbar button with the page action ( https://bugzilla.mozilla.org/show_bug.cgi?id=1366041 ) which is a separate bug.

> If this item is in addition to some photon-specific "take a shot" UI, I'm not sure what the flow should be. If a user hasn't tried Screenshots before, I think we'd want to show them the onboarding flow, and maybe not open a link to the shots page. @johngruen, thoughts?
> 
> I'm also not sure we'd want to reuse the same icon for multiple commands related to the same feature (take a shot vs visit shots page).

That also works for me, I think it's a discussion to have with Aaron Benson - I'm just doing what the mock says to do right now. We might even just want to not show the screenshots item until we know the user has taken at least 1 screenshot (but I don't know how to determine that, or to determine we need to show the different icon / different page. Presumably the website could just show the onboarding flow if it's opened and the user has no shots? That would be the simplest from my perspective...